web_ui/server_api.go

```diff
@@ -197,10 +240,20 @@ func HandleCreateDowntime(ctx *gin.Context) {
 		EndTime:     downtimeInput.EndTime,
 	}
 
-	if err := database.CreateDowntime(&downtime); err != nil {
-		ctx.JSON(http.StatusInternalServerError, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to create downtime: " + err.Error()})
+	// Mirror to Registry when running as Origin/Cache so downtime persists centrally (Director polls Registry for all sources)
+	if err := mirrorDowntimeToRegistry(ctx, downtime, http.MethodPost, idStr); err != nil {
+		ctx.JSON(http.StatusBadGateway, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to mirror downtime to registry: " + err.Error()})
 		return
 	}
+
+	if err := database.CreateDowntime(&downtime); err != nil {
+		if errors.Is(err, gorm.ErrDuplicatedKey) {
+			log.Debugf("Downtime already exists: %v; skipping creation", downtime)
+		} else {
+			ctx.JSON(http.StatusInternalServerError, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to create downtime: " + err.Error()})
+			return
+		}
+	}
```

Comment:
I think this is the section we talked about changing in a follow-up PR -- the Origin/Cache should not rely on a response from the Registry to populate its own downtime table.

In the meantime, in case we don't get to that PR for the upcoming release, can you add a comment before this section that explicitly states why this code waits for a registry response before recording the downtime? That way if this doesn't change right away, we still have a comment explaining why you chose this design.